python/tank/descriptor/descriptor.py

```diff
@@ -30,7 +30,9 @@ def create_descriptor(
         fallback_roots=None,
         resolve_latest=False,
         constraint_pattern=None,
-        local_fallback_when_disconnected=True):
+        local_fallback_when_disconnected=True,
+        config_app_store_proxy=None
```

Comment:
I really don't like having to do this, and it probably needs a better solution. The AppStore descriptors looks up the  proxy settings by looking inside the config/core/shotgun.yml, which can't be found in a path relative to the core anymore. I decided to push the value through the API, but that's super error prone and we can't expect clients to remember to pass this through, so a better solution needs to be considered.

Reply:
It should also been mentionned that by passing the parameter in this way, we're losing something.

There are the states for the setting the in the file
- missing
- empty
- set

But in Python, when passing `None` or a host url/ip, **missing** and **empty** are expressed with a `None` and yet have different behaviour. A **missing** value in Shotgun.yml means we fall back on the http_proxy value, while an **empty** value means we're forcing the code to not use a proxy. This is why some connection relation tests are failing right now.